staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/helpers.go

```diff
@@ -232,3 +232,14 @@ func HasObjectMetaSystemFieldValues(meta Object) bool {
 	return !meta.GetCreationTimestamp().Time.IsZero() ||
 		len(meta.GetUID()) != 0
 }
+
+// ResetObjectMetaForStatus forces the meta fields for a status update to match the meta fields
+// for a pre-existing object. Status updates are for updating status, not metadata.
+func ResetObjectMetaForStatus(meta, existingMeta Object) {
+	meta.SetGeneration(existingMeta.GetGeneration())
+	meta.SetSelfLink(existingMeta.GetSelfLink())
+	meta.SetLabels(existingMeta.GetLabels())
+	meta.SetAnnotations(existingMeta.GetAnnotations())
```

Comment:
For better or ill, our existing status handlers allow changing labels and annotations. At least last I checked.

Reply:
> our existing status handlers allow changing labels and annotations. 

@lavalamp Fwiw, the idea is to only enable this for new resources (like CRDs, custom resources, etc.. which don't allow changing labels and annotations now).